openupgrade_scripts/scripts/analytic/16.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,84 @@
+from openupgradelib import openupgrade
+
+_fields_renames = [
+    (
+        "account.analytic.account",
+        "account_analytic_account",
+        "group_id",
+        "plan_id",
+    ),
+    (
+        "account.analytic.line",
+        "account_analytic_line",
+        "group_id",
+        "plan_id",
+    ),
+]
+_models_renames = [("account.analytic.group", "account.analytic.plan")]
+_tables_renames = [("account_analytic_group", "account_analytic_plan")]
+
+
+def analytic_account_set_group_id_if_null(env):
+    """
+    We would like to fill for group_id column before start renaming it into plan_id
+    """
+    openupgrade.logged_query(
+        env.cr,
+        """
+        WITH tmp AS (
+            SELECT id, company_id
+            FROM account_analytic_account
+            WHERE group_id IS NULL
+        ),
+        inserted_group AS (
+            INSERT INTO account_analytic_group (name, company_id)
+            SELECT 'Dummy analytic group', company_id
+            FROM tmp
+            LIMIT 1
+            RETURNING id
+        )
+        UPDATE account_analytic_account aaa
+        SET group_id = (
+            SELECT id
+            FROM inserted_group
+            LIMIT 1
+        )
+        WHERE aaa.id IN (SELECT id FROM tmp);
```

Comment:
@duong77476 chỗ này chỉ cần tạo 1 plan tất cả các tài khoản là được, do trường công ty không bắt buộc
kiểu của nó thì đặt là `unavailable` để tránh lọc ra
Note: nhớ tạo cột `default_applicability` cho nó
```suggestion
        WITH inserted_group AS (
            INSERT INTO account_analytic_group (name, default_applicability)
            SELECT 'Dummy Analytic Plan', 'unavailable'
            RETURNING id
        )
        UPDATE account_analytic_account aaa
        SET group_id = (
            SELECT id
            FROM inserted_group
            LIMIT 1
        )
        WHERE aaa.group_id IS NULL;
```